merge_sort/Cpp/merge_sort.cpp

```diff
@@ -0,0 +1,73 @@
+#include <iostream>
+#include <cstdio>
+#include <vector>
+using namespace std;
+std::vector<int> vec;
+
+void join(std::vector<int>& arr, int, int, int);
+void sort(std::vector<int>& arr, int, int);
+void show(std::vector<int>& arr);
+
+int main() {
+	int n = 10, a, i, arr[] = {2, 4, 3, 6, 5, 1, 9, 0, 8, 7};
+	// scanf("%d", &n); //For custom input
+	// printf("Enter %ld number (space separated): ", n); //For custom inup
+	for (i = 0; i < n; i++) {
+		//scanf("%d", &a);
+		vec.push_back(arr[i]); //vec.push_back(a); // For custom input
+	}
+	a = 0;
+	sort(vec, a, n-1);
+	show(vec);
+	return 0;
+}
+
+void join (std::vector<int>& arr, int l, int m, int r) {
+	//  merge operation of merge sort.
+	int i, j, k;
+	int s1 = m-l+1, s2 = r-m;
+	int L[s1], R[s2];
+	for (i = 0; i < s1; i++)
+		L[i] = arr[l+i];
+	for (j = 0; j < s2; j++)
+		R[j] = arr[m+j+1];
+	i = 0; j = 0; k = l;
+	while (i < s1 && j < s2) {
+		if (L[i] < R[j]) {
+			arr[k] = L[i];
+			i++;
+		}
+		else {
```

Comment:
An else should appear on the same line as the preceding }  [whitespace/newline] [4]

*Origin: CPPLintBear, Section: `all.cpplint`.*